src/web_app_skeleton/docker/dev_entrypoint.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# This is the entrypoint script used for development docker workflows. By
+# default it will:
+#  - Install shards.
+#  - Run migrations.
+#  - Start the dev server.
+# It also accepts any commands to be run instead.
+
+
+warnfail () {
+  echo "$@" >&2
+  exit 1
+}
+
+case ${1:-} in
+  "") # If no arguments are provided, start lucky dev server.
+    ;;
+
+  *) # If any arguments are provided, execute them instead.
+    exec "$@"
+esac
+
+if ! [ -d bin ] ; then
+  echo "Creating bin directory"
+  mkdir bin
```

Comment:
does this need the `-p` flag?

Reply:
It shouldn't. It's only one level deep (rather than `mkdir one/two/three/bin`) and it'll only get run if the bin directory doesn't already exist.

I honestly don't think this stanza should need to be in here. The compiler should be creating this folder when the binary is created, but it doesn't. For reasons I don't understand the `ld` step  fails because the bin directory doesn't exist. This may be a linux problem I'm unaware of, but I was able to work around it with this.